tracing-attributes/src/lib.rs

```diff
@@ -109,12 +109,16 @@ mod expand;
 ///
 /// Additional fields (key-value pairs with arbitrary data) can be passed to
 /// to the generated span through the `fields` argument on the
-/// `#[instrument]` macro. Strings, integers or boolean literals are accepted values
-/// for each field. The name of the field must be a single valid Rust
-/// identifier, nested (dotted) field names are not supported.
+/// `#[instrument]` macro. Strings, integers or boolean literals are accepted
+/// values for each field. The name of the field may be a nested (dotted) field
+/// name.
+///
+/// When an additional field does not include a value,
+/// [`Empty`](tracing-core::field::Empty) will be inserted and this field can be
+/// recorded later with [`Span::record`](tracing::Span::record).
 ///
 /// Note that overlap between the names of fields and (non-skipped) arguments
-/// will result in a compile error.
+/// will result in recording the value provided to the macro.
```

Comment:
I would add something like "and ignore the value provided to the function." to the sentence as this could be a slightly unpleasant surprise.

Reply:
Done. I've also changed the paragraph just before because I've noticed it might also be misleading out of context.